solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -750,6 +753,9 @@ public void load() {
 
     zkSys.initZooKeeper(this, cfg.getCloudConfig());
     if (isZooKeeperAware()) {
+      //initialize ZkClient metrics
+      zkSys.getMetricsProducer().initializeMetrics(solrMetricsContext, "zkClient");
```

Comment:
since all these metrics are tracked in and directly relevant to SolrZkClient, could we just remove all changes to ZkContainer and have SolrZkClient directly implement SolrMetricProducer, initialized directly here via:
```java
zkSys.getZkController().getZkClient().initializeMetrics(solrMetricsContext, "zkClient");
```

Reply:
`SolrMetricProducer` is not visible to classes in SolrJ